### ISSUE TITLE
OTEL_EXCLUDED_HTTP_URLS to ignore health checks

### ISSUE THREAD

**Issue by sarasensible** [open-telemetry/opentelemetry-specification#3057]
We currently have the OTEL_PYTHON_EXCLUDED_URLS environment variable for Python that allows us to ignore health checks on an auto-injected container.  However we do not have an environment variable for Node JS.  It seems to me this should be standardized and added since auto-injection does not allow for something like adding the ignoreIncomingRequestHook option of [https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-instrumentation-http/README.md#http-ins[…]n-options](https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-instrumentation-http/README.md#http-instrumentation-options) which is the current solution to this problem.

**Comment by dyladan** [comment]
I support this feature request. Also happy to prototype in JS if needed. 

It is my understanding that sdk maintainers are expected to wait for spec to implement env vars but if that isn't the case can a tc member please let me know.

**Comment by tsloughter** [comment]
This shouldn't be an SDK feature, should it? I was putting these in instrumentation libraries. So nothing to wait on the spec for, even tho it certainly would be useful to have a common spec for common filters like HTTP.

**Comment by dyladan** [comment]
I hadn't really considered that they could be in instrumentation without spec. Is that explicitly allowed?

**Comment by Aneurysm9** [comment]
> This shouldn't be an SDK feature, should it? I was putting these in instrumentation libraries.

I'm not sure it belongs in instrumentation libraries, either.  Most generally this could be a sampler feature that can be used to ensure spans with certain properties are not sampled.  This can be implemented as a sampler that wraps another sampler to avoid re-implementation of the functionality and preserve the ability to use it with custom samplers.

> it certainly would be useful to have a common spec for common filters like HTTP.

Having filters implemented by instrumentation would be another option.  We have used this in [some Go instrumentation](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.37.0/filters), but that becomes tightly coupled to HTTP and requires [a different interface](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.37.0/filters) for gRPC or any other transport.  Given that the configuration surface of these filters is considerably more complex than a list of URLs to exclude, I wouldn't want to try to specify this functionality until we have a generalized configuration system in place.

**Comment by dyladan** [comment]
> > This shouldn't be an SDK feature, should it? I was putting these in instrumentation libraries.
> 
> I'm not sure it belongs in instrumentation libraries, either. Most generally this could be a sampler feature that can be used to ensure spans with certain properties are not sampled. This can be implemented as a sampler that wraps another sampler to avoid re-implementation of the functionality and preserve the ability to use it with custom samplers.

I fully agree with this and is the reason I initially encouraged @sarasensible to create this issue.

> > it certainly would be useful to have a common spec for common filters like HTTP.
> 
> Having filters implemented by instrumentation would be another option. We have used this in [some Go instrumentation](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp@v0.37.0/filters), but that becomes tightly coupled to HTTP and requires [a different interface](https://pkg.go.dev/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.37.0/filters) for gRPC or any other transport. Given that the configuration surface of these filters is considerably more complex than a list of URLs to exclude, I wouldn't want to try to specify this functionality until we have a generalized configuration system in place.

There should probably be a spec for _how_ to create instrumentation specific environment variables (there might already be) so that they don't collide with names specified for the SDK in some future. If an instrumentation implemented `OTEL_JAVASCRIPT_EXCLUDED_URLS` and the spec later decided to implement it as a sampler configuration env var then we could have a problem.

**Comment by tigrannajaryan** [comment]
@open-telemetry/configuration-maintainers please take a look and assign appropriately.

**Comment by douglasg14b** [comment]
Any update on this?
